src/predicate.rs

```diff
@@ -185,6 +309,63 @@ impl<A: Predicate, B: Predicate> Predicate for Descendant<A, B> {
                 node = parent;
             }
         }
-        return false;
+        false
+    }
+}
+
+/// Matches if inner Predicate `B` matches the node and `A` matches the node before this one.
+///
+/// This is equivalent to the `before + after` CSS selector.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct ImmediatelyAfter<A, B>(pub A, pub B);
```

Comment:
The CSS selector is officially called the Adjacent Sibling Combinator. What do you think about naming this `AdjacentSibling`?

Reply:
Will do.